src/curobo_rviz.cpp

```diff
@@ -58,26 +56,87 @@ namespace curobo_rviz
+        }
+        
+        // set parameters on parameter server
+        param_client_->set_parameters_atomically({rclcpp::Parameter("timeout", timeout_)});
         RCLCPP_INFO(node_->get_logger(), "Timeout set to %.2f", timeout_);
     }
 
     void RvizArgsPanel::updateTimeDilationFactor(double value)
     {
         time_dilation_factor_ = value;
+        while (!param_client_->wait_for_service(std::chrono::seconds(1))) {
+            RCLCPP_INFO(node_->get_logger(), "service not available, waiting again...");
+        }
+        
+        // set parameters on parameter server
+        param_client_->set_parameters_atomically({rclcpp::Parameter("time_dilation_factor", time_dilation_factor_)});
         RCLCPP_INFO(node_->get_logger(), "Time dilation factor set to %.2f", time_dilation_factor_);
     }
 
+    void RvizArgsPanel::updateVoxelSize(double value)
+    {
+        voxel_size_ = value;
+        RCLCPP_INFO(node_->get_logger(), "Voxel size changed to %.2f", voxel_size_);
+    }
+
+    void RvizArgsPanel::updateCollisionActivationDistance(double value)
+    {
+        collision_activation_distance_ = value;
+        RCLCPP_INFO(node_->get_logger(), "Collision activation distance changed to %.2f", collision_activation_distance_);
+    }
+
+    void RvizArgsPanel::on_confirmPushButton_clicked()
+    {
+        RCLCPP_INFO(node_->get_logger(), "Confirm button clicked.");
+        while (!param_client_->wait_for_service(std::chrono::seconds(3))) {
+            RCLCPP_INFO(node_->get_logger(), "service not available, waiting again...");
+        }
+        
+        // set parameters on parameter server
+        param_client_->set_parameters_atomically({rclcpp::Parameter("voxel_size", voxel_size_),
+                                       rclcpp::Parameter("collision_activation_distance", collision_activation_distance_)});
+        RCLCPP_INFO(node_->get_logger(), "Parameters set:\voxel_size: %.2f, collision_activation_distance: %.2f", voxel_size_, collision_activation_distance_);
+        
+        // call Trigger service update_motion_gen_config
+        rclcpp::Client<std_srvs::srv::Trigger>::SharedPtr client = node_->create_client<std_srvs::srv::Trigger>("/curobo_gen_traj/update_motion_gen_config");
+        auto request = std::make_shared<std_srvs::srv::Trigger::Request>();
+
+        while (!client->wait_for_service(std::chrono::seconds(1))) {
+            RCLCPP_INFO(node_->get_logger(), "service not available, waiting again...");
+        }
+        auto result = client->async_send_request(request);
+        RCLCPP_INFO(node_->get_logger(), "Service call sent.");
+        
+        if (rclcpp::spin_until_future_complete(node_, result) == rclcpp::FutureReturnCode::SUCCESS) {
+            RCLCPP_INFO(node_->get_logger(), "Service call successful.");
+        } else {
+            RCLCPP_ERROR(node_->get_logger(), "Service call failed.");
+        }
```

Comment:
The [std_srvs/Trigger](http://docs.ros.org/en/noetic/api/std_srvs/html/srv/Trigger.html)'s response contains the `success` and `message` arguments. That information could be used in the logging for more accuracy. For example, the `success` argument could be `false`, but the `rclcpp::FutureReturnCode::SUCCESS` argument will be `true`, since that only checks that the service responded, not that it didn't get an exception in its execution.